test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -220,6 +227,7 @@
     private Function<Client, Client> clientWrapper;
 
     public InternalTestCluster(String nodeMode, long clusterSeed, Path baseDir,
+                               int numMasterNodes,
```

Comment:
I have checked and I think we only ever use two values for numDedicatedMasterNodes, `0` to have none of them, and `-1` to randomize the behaviour (default). I think having a boolean in the annotation would clarify things a bit. true would mean random behaviour, you may have dedicated master nodes. false means no dedicated masters. Then the interaction with numDataNodes is probably a bit clearer as well. Could be further improved but I think this is a quick win at the moment. What do you think?